msr-flows/msbr_compare/saltproc_read.py

```diff
@@ -308,14 +310,192 @@ def evaluate(time, hdf5_path, fuel_path):
+        raise Exception
+
+    db = tb.open_file(db_file)
+    pre_fuel = db.root.core_adensity_before_reproc
+    post_fuel = db.root.core_adensity_after_reproc
+    pre_atoms = list()
+    post_atoms = list()
+    removal_mass = list()
+    possible_times = [3 * x for x in range(0, 2270)]
+    for time_index in range(len(possible_times)):
+        comp_pre = pre_fuel[time_index, :]
+        atoms_pre = comp_pre[iso_index]
+        comp_post = post_fuel[time_index, :]
+        atoms_post = comp_post[iso_index]
+        pre_atoms.append(atoms_pre)
+        post_atoms.append(atoms_post)
+        # mol/cc -> kg/cc Divide by 3000 b/c 3 day step and 1000 kg/g
+        removed = abs(atoms_pre - atoms_post) * 4.871E7 * \
+            mol_weight / (ui.SP_step_size * 1000)
+        removal_mass.append(removed)
+    start_index = possible_times.index(ui.start_time)
+    end_index = possible_times.index(ui.end_time)
+    print(f'Average removal per day of {iso}')
+    print(f'Total average: {np.mean(removal_mass)} kg/day')
+    print('Plotting average vs actual over time')
+    plt.plot(possible_times, removal_mass)
+    plt.xlabel('Time [d]')
+    plt.ylabel('Refill Rate [kg/day]')
+    plt.ylim(0, 3)
+    plt.savefig(f'{iso}rem_massv01.png')
+    plt.close()
+    removal_mass = removal_mass[start_index:end_index]
+    avg_kg_day = np.mean(removal_mass)
+    print(f'{avg_kg_day} kg per day')
+    db.close()
+    return avg_kg_day
+
+
+def check_total_mass(db_file):
+    """
+    Iterate through every isotope in each time step
+        and generate a plot of the net mass difference each step
+
+    Parameters
+    ----------
+    db_file : str
+        Path to the database hdf5 file.
+
+    Returns
+    -------
+    None
+
+    """
+
+    db = tb.open_file(db_file, mode='r')
+    pre_fuel = db.root.materials.fuel.before_reproc.comp
+    isomap = pre_fuel.attrs.iso_map
+    sim_param = db.root.simulation_parameters
+    day_eds = [x['cumulative_time_at_eds'] for x in sim_param.iterrows()]
+    fuel_after = db.root.materials.fuel.after_reproc.comp
+    isomap = fuel_after.attrs.iso_map
+    mass_over_time = list()
+    time_list = list()
+    prev_mass = 0
+    for dts in range(len(day_eds)):
+        net_mass_cur_time = 0
+        composition_at_time_after = fuel_after[dts, :]
+        for iso in isomap:
+            net_mass_cur_time += composition_at_time_after[isomap[iso]]
+        if dts == 0:
+            mass_over_time.append(0)
+        else:
+            mass_over_time.append(net_mass_cur_time - prev_mass)
+        prev_mass = net_mass_cur_time
+    db.close()
+    plt.plot(day_eds, mass_over_time)
+    plt.xlabel('Time [d]')
+    plt.ylabel('Mass [g]')
+    plt.savefig('SaltProc_netmass.png')
+    plt.close()
+
+    return
+
+
 if __name__ == '__main__':
 
     fuel_input_path = './ss-data-test/mat_prepr_comp_geo_1_boc.ini'
-    hdf5_input_path = './ss-data-test/6000_day_SS_data'
+    hdf5_input_path = './ss-data-test/db_saltproc_2270.hdf5'
```

Comment:
That's a good idea, I haven't put the data anywhere yet since I was still generating good results. Would it be better to put the data on GitHub or in the group box?